### ISSUE TITLE
Non-superusers can't change hidden status of languages

### ISSUE THREAD

**Issue by adrianbj** [processwire/processwire-issues#1409]
### Short description of the issue

I know this isn't a common use case, but users with `lang-edit` can't change the hidden status of languages. They can see and change the unpublished status, but in my use case I want to set languages to hidden for a period of time before they become live - note that I am using them for controlling languages available for a frontend language selection field.

### Expected behavior

Hidden checkbox is available on the Settings tab.

### Actual behavior

It's not shown

**Comment by daun** [comment]
Isn't the problem with hiding languages that it will also make them disappear from the edit screen? So you can't start filling in the content before they become live either way.

**Comment by adrianbj** [comment]
@daun - perhaps, but I don't have any multi-language fields anyway. I use the languages for triggering a runtime translation of an SMS message sent to the user. I am using the hidden status to remove it from a frontend form where the user edits their profile. I could do this other ways, eg add a new "hide" checkbox field to the language template, but I don't really understand if it's an intentional decision by @ryancramerdesign to make this unavailable for non-superusers, or a bug - seems like a bug to me.

**Comment by daun** [comment]
I agree it seems like a bug as far as expected vs. actual behavior go.

**Comment by matjazpotocnik** [comment]
@adrianbj does it help if the user has page-hide permission?

**Comment by adrianbj** [comment]
@matjazpotocnik - I'm not sure, but the user has the `page-edit` permission and they can edit languages. Given that I don't have the `page-hide` permission installed, it should be delegated to `page-edit` (https://github.com/processwire/processwire/blob/6ff498f503db118d5b6c190b35bd937b38b80a77/wire/core/Permissions.php#L43) so I feel like that should work.

**Comment by matjazpotocnik** [comment]
I have a role editor1 with lang-edit and page-edit perms (I also don't have page-hide perm), and this user has the option to hide a language in the Settings tab:

![image](https://github.com/processwire/processwire-issues/assets/7663992/e5c01f30-2db4-47a1-884b-0c5b1c611017)

I'm obviously missing something...

**Comment by adrianbj** [comment]
Weird, in my case, the user has the `lang-edit` permission and this is what they see. I wonder if it might be because my users are on an alternate user template / parent page. That setup still seems to break things in various obscure ways.

<img width="363" alt="image" src="https://github.com/processwire/processwire-issues/assets/871211/155fb33f-4d3c-4b36-8763-afbbd099894f">

**Comment by matjazpotocnik** [comment]
I tested with the user in an alternate branch, and it's working here... Can you test with the "regular" user first? Just to rule out "alternate" user issue?

**Comment by adrianbj** [comment]
OK, on another install where all the users are regular, I actually see this where there are no status options to change.

<img width="348" alt="image" src="https://github.com/processwire/processwire-issues/assets/871211/ed003ab9-c153-4f8b-a121-8bc1a1ff541d">

**Comment by matjazpotocnik** [comment]
Ok, I'm out of ideas, I can't debug if I can't reproduce...

**Comment by adrianbj** [comment]
For sanity's sake I just tested another install where initially no users had language editing permission. Gave it to them, and tested and I see the same empty Settings section again. These are all very different sites and I can't think of anything that would cause their behavior to be different to what you're seeing. Maybe Ryan has some thoughts?

**Comment by matjazpotocnik** [comment]
It must be something with roles/permissions... my test user has this permission in their role: page-view, page-edit, lang-edit, page-delete, page-move, profile-edit, tracy-debugger

**Comment by matjazpotocnik** [comment]
@adrianbj, can we close this?

**Conclusion.** This is not a defect in `ProcessLanguage` or `ProcessPageEdit`. It is the intended consequence of ProcessWire's permission model: *once a finer-grained permission is installed, it stops falling back to its coarser delegate.* Granting `page-edit` only covers `page-hide` while `page-hide` remains uninstalled. Admins who rely on the delegation should refrain from installing `page-hide` (keeping it undeployed lets `page-edit` cover it), or alternatively assign `page-hide` to the relevant role.

The hidden checkbox on the language Settings tab is controlled by `ProcessPageEdit::getAllowedStatuses()`, which gates it behind `$user->hasPermission('page-hide', $page)`.

Then in `User::___hasPagePermission()`, the `page-hide → page-edit` delegation is only honoured **while `page-hide` is not installed**:

```php
} else if(!$permissions->has($name)) {
    …
    $p = isset($delegated[$name]) ? $delegated[$name] : $name;
}
```

Once `page-hide` exists as a permission row, `Permissions::getDelegatedPermission()` returns `''` and the fallback to `page-edit` disappears — the user then needs `page-hide` outright.

Concretely:
- Site without `page-hide` installed → `page-edit` suffices → checkbox shows (matjazpotocnik's case, and mine).
- Site with `page-hide` installed → `page-edit` no longer covers it → checkbox gone (adrianbj's recurring case).

As a pragmatic fix on the affected sites, either grant the role the `page-hide` permission, or leave `page-hide` uninstalled so `page-edit` keeps covering it.

**Comment by adrianbj** [comment]
I don't have the `page-hide` permission installed.

But I am still seeing this:

<img width="286" height="234" alt="Image" src="https://github.com/user-attachments/assets/95a5d6c1-5656-4184-8530-e16c027fd192" />

**Comment by adrianbj** [comment]
I finally got to the bottom of this, and it's not related to whether `page-hide` is installed (it's not installed on any of my affected sites). It's an asymmetry between how `editable()` and context-aware `hasPermission()` treat language pages.

### Root cause

The Hidden checkbox is gated by `$user->hasPermission('page-hide', $this->page)` in `ProcessPageEdit::getAllowedStatuses()`. Two code paths then diverge for language pages:

1. `PagePermissions::___pageEditable()` has an explicit special case for languages:

```php
// special conditions apply if the page is a Language
if($page->template == 'language' && $user->hasPermission('page-edit') && $user->hasPermission('lang-edit')) {
    return $this->hasLangPermission("page-edit-lang-$page->name", $user);
}
```

Note both permission checks are made **without page context**. This is why `lang-edit` users can edit languages at all, and why the Unpublished checkbox shows — `publishable()` falls back to `editable()` when `page-publish` isn't installed.

2. `User::___hasPagePermission('page-hide', $languagePage)` has no such special case. The delegation of `page-hide` to `page-edit` works fine, but the check then walks tree-based access:

```php
// all page- permissions except page-view and page-add require page-edit access on $page, so check against that
if(strpos($name, 'page-') === 0 && $name != 'page-view' && $name != 'page-add') {
    if($accessTemplate && !in_array($role->id, $accessTemplate->editRoles)) continue;
}
```

The `language` template doesn't use roles, so access resolves up the admin tree — `$languagePage->getAccessTemplate('page-edit')` returns the `admin` template, whose `editRoles` is empty. Every role fails that check, so the checkbox never renders. Same for `page-lock`.

Verified on one of my affected installs with a user whose role has `page-edit` + `lang-edit`:

```php
$lang->editable();                            // true
$lang->publishable();                         // true
$user->hasPermission('page-hide');            // true (no context — delegation to page-edit works)
$user->hasPermission('page-hide', $lang);     // false (page context — tree access fails)
```

### Why @matjazpotocnik sees different behavior

It depends entirely on which template the language page *inherits access from*, and whether that template's `editRoles` happens to include the test user's role. Nothing to do with languages or with `page-hide` being installed — matjaz, I'd guess on your test install the access either resolves to a template where your editor role has edit access, or your admin template defines edit roles.

### Suggested fix

Mirror the language special case from `PagePermissions::___pageEditable()` in `User::___hasPagePermission()` — when `$page->template == 'language'`, grant page-* permissions from the role level (plus `lang-edit`) instead of walking tree-based access, since language pages inherit access from the admin tree where non-superusers never have edit roles.

### Workaround

For anyone else hitting this, a site-level hook works because `page-hide` is honored again once `hasPagePermission()` returns true (the same gate controls both rendering and saving of the checkbox):

```php
$wire->addHookAfter('User::hasPagePermission', function(HookEvent $event) {
    if($event->return) return;
    $name = $event->arguments(0);
    if($name !== 'page-hide' && $name !== 'page-lock') return;
    $page = $event->arguments(1);
    if(!$page instanceof Page || $page->template->name !== 'language') return;
    $u = $event->object;
    if($u->hasPermission('lang-edit') && $u->hasPermission($name)) $event->return = true;
});
```
